File: 104-Maximum-Depth-of-Binary-Tree.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    # Recursive Solution
    def maxDepth(self, root: TreeNode) -> int:
        if not root:
            return 0
        return max(self.maxDepth(root.left)+1, self.maxDepth(root.right)+1)

    # Iterative Solution
    def maxDepth(self, root):
        maxDepth = 0
        if root:
            currLevel = [root]
        else:
            currLevel = []
        while currLevel:
            maxDepth += 1
            q = []
            for val in currLevel:
                if val.left:
                    q.append(val.left)
                if val.right:
                    q.append(val.right)
            currLevel = q
        return maxDepth

File: JavaScript/104-Maximum-Depth-of-Binary-Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root) {
    if (!root) return 0
    return Math.max(maxDepth(root.left) + 1, maxDepth(root.right) + 1)
};


var maxDepth = function (root) {
    ans = 0
    if (root) {
        lvl = [root]
    } else lvl = []

    while (lvl.length > 0) {
        ans += 1
        queue = []
        for (var i = 0; i < lvl.length; i++) {
            if (lvl[i].left) queue.push(lvl[i].left)
            if (lvl[i].right) queue.push(lvl[i].right)
        }
        lvl = queue
    }
    return ans
};